lib/pluginmanager/util.rb

```diff
@@ -17,6 +21,7 @@ def self.logstash_plugin?(plugin, version = nil)
       end
     else
       dep = Gem::Dependency.new(plugin, version || Gem::Requirement.default)
+      Gem.sources = Gem::SourceList.from(options[:rubygems_source]) if options[:rubygems_source]
```

Comment:
@ph agree, we are due for a 3rd refactor here. Do we have an issue for this idea?